### PR TITLE
replace engine and framework mirror args with github username arg

### PR DESCRIPTION
**Pull request by itsjustkevin**
Remove the `--framework-mirror` and `--engine-mirror` args and replace them with the `--github-username` arg which interpolates the username into the framework and engine mirror urls.

### CODE REVIEW

**Comment by christopherfujino** [comment]
this is fine as long as you're willing to accept the constraint that it will only ever work with a project name that matches the name of the canonical upstreams.

**Comment by itsjustkevin** [comment]
@christopherfujino do you feel that this solution isn't generic enough?  

I am looking at this from the perspective of the people currently using the release tooling, but I can absolutely see where this could be an issue down the line if we decide to make changes.

**Comment by christopherfujino** [comment]
> @christopherfujino do you feel that this solution isn't generic enough?
> 
> I am looking at this from the perspective of the people currently using the release tooling, but I can absolutely see where this could be an issue down the line if we decide to make changes.

As a power user, I've definitely used the conductor on secondary/tertiary forks. It doesn't happen often, though.

File: dev/conductor/core/lib/src/start.dart [hunk 1]
```diff
@@ -130,21 +128,19 @@ class StartCommand extends Command<void> {
...
       kEngineUpstreamOption,
       argumentResults,
       platform.environment,
     )!;
-    final String engineMirror = getValueFromEnvOrArgs(
-      kEngineMirrorOption,
-      argumentResults,
-      platform.environment,
-    )!;
+    final String engineMirror = 'https://github.com/$githubUsername/engine.git';
```
[the remark this answers is not in the document]
Comment (itsjustkevin) [reply dev/conductor/core/lib/src/start.dart:143]: If that is the case, we could hardcode the mirrors and remove the argument completely.  Do you see another approach I could take here?

**Comment by itsjustkevin** [comment]
> > @christopherfujino do you feel that this solution isn't generic enough?
> > I am looking at this from the perspective of the people currently using the release tooling, but I can absolutely see where this could be an issue down the line if we decide to make changes.
> 
> As a power user, I've definitely used the conductor on secondary/tertiary forks. It doesn't happen often, though.

Thanks for pointing this out, I have not considered this use case.  Open for suggestions.

File: dev/conductor/core/test/start_test.dart [hunk 2]
```diff
@@ -212,7 +194,12 @@ void main() {
           command: <String>['git', 'add', '--all'],
         ),
         const FakeCommand(
-          command: <String>['git', 'commit', '--message', 'Update Dart SDK to $nextDartRevision'],
+          command: <String>[
+            'git',
+            'commit',
+            '--message',
+            'Update Dart SDK to $nextDartRevision'
```
Comment (christopherfujino) [inline dev/conductor/core/test/start_test.dart:201]:
```suggestion
            'Update Dart SDK to $nextDartRevision',
```

**Review by christopherfujino — COMMENTED** [review]
Generally, most of the dart format changes here are against the repo's style guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#formatting

**Comment by itsjustkevin** [comment]
> Generally, most of the dart format changes here are against the repo's style guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#formatting

Do you have a VSCode settings.json file I could use to avoid this in the future?  It would be nice if there was one available, there may be one already somewhere.

**Comment by christopherfujino** [comment]
> > Generally, most of the dart format changes here are against the repo's style guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#formatting
> 
> Do you have a VSCode settings.json file I could use to avoid this in the future? It would be nice if there was one available, there may be one already somewhere.

It's trivial to globally disable autoformat on an editor level (https://linuxpip.org/vscode-format-on-save/#:~:text=Save%20not%20working-,Enable%2FDisable%20Format%20On%20Save,click%20on%20the%20right%20option.). I'm sure you could also do this on a language level for all Dart source files, but I'm not sure how to do it on a per-repo level (not sure if you submit patches to flutter/cocoon).

**Comment by itsjustkevin** [comment]
> Generally, most of the dart format changes here are against the repo's style guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#formatting

> It's trivial to globally disable autoformat on an editor level (https://linuxpip.org/vscode-format-on-save/#:~:text=Save%20not%20working-,Enable%2FDisable%20Format%20On%20Save,click%20on%20the%20right%20option.). I'm sure you could also do this on a language level for all Dart source files, but I'm not sure how to do it on a per-repo level (not sure if you submit patches to flutter/cocoon).

Will update after reading the style guide fully.

File: dev/conductor/core/lib/src/start.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (itsjustkevin) [reply dev/conductor/core/lib/src/start.dart:143]: Looking at this comment again:  What if the release team did share a common github account?  We would probably still want some kind of mechanism to verify who is doing what under that shared account. 

Will look further into this.

**Review by christopherfujino — APPROVED** [review]
LGTM

File: dev/conductor/core/test/start_test.dart [hunk 3]
```diff
@@ -884,13 +935,27 @@ void main() {
       expect(state.releaseType, ReleaseType.STABLE_INITIAL);
     });
 
-    test('StartContext gets engine and framework checkout directories after run', () async {
+    test('mirrors are interpreted from github username', () async {
```
Comment (christopherfujino) [inline dev/conductor/core/test/start_test.dart:938]: I don't think you need this test--it's not actually testing any code under lib/src.

Comment (christopherfujino) [reply dev/conductor/core/test/start_test.dart:938]: If you want to test the interpolation, you can hard-code the `engineMirror` and `frameworkMirror` strings on lines 26 & 28, and the `FakeCommand`'s will verify that the conductor is interpolating them correctly.

Comment (itsjustkevin) [reply dev/conductor/core/test/start_test.dart:938]: Yeah, this is the test I was using to figure out what was going on, it can be removed.

**Review by itsjustkevin — COMMENTED** [review]
removed

File: dev/conductor/core/lib/src/start.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (CaseyHillers) [reply dev/conductor/core/lib/src/start.dart:143]: Exactly :-)

We could see if https://github.com/fluttergithubbot is ok to be used. We may need to lock it down so not all maintainers are able to contribute to the forks.

[the remark this answers is not in the document]
Comment (itsjustkevin) [reply dev/conductor/core/lib/src/start.dart:143]: Opened https://github.com/flutter/flutter/issues/109353 to track this.

**Pull request merged**